---
title: "Synbio Notes"
date: 2023-12-23
tags: synbio
---

== dec 24

=== introduction

*defining synbio*: 
making biology more modular, robut, simulation, prototyping, standardization, more accessble to the engineering discpline. 
treating biology like a system. design and construction of new biologicla parts, devices and systems.
the redesign of existing natural biological systems for useful purposes. the endeavor to make biology and it's principles accessible to engineers.

*through history*
molecular biology -> molecular biotechnology -> synthethic biology
plasmids: transfer useful properties to e.coli

1. discovery of DNA structure and role in heredity (fundamental science at the moledular level)
2. development of plasmids, systems biology, Sanger DNA Sequencing (fundamental and applied science and the molecular level, fundamantal science at systems level)
3. DNA synthesis, geneti circuits, redesigning phage genome (fundamental and applied science at moledular and system level)

two timelines: 

1. palsson timeline: focus on early developments and synthethic cells

. fundamental basis of synbio (bottom up)
. putting life together -> lets use understand it better 
	. minimalist cells and synthethic chromosomes enable fundamental discoveries about roles genes play in cell viabiliy.

2. collins timeline: focus on development of genetic tools and their applications

. putting life together -> lets us use it better
	. genetic tools let us manipulate cells with more control

DNA synthesis and sequencing important for both

*through purpose*
an engineering discipline -> social, ethical and global implications and financial and commerial applications 


*study and design life to discover truths and benefit society, humans and environment.*
if this is the purpose of synbio, there must be judgement made by governing bodies to regulate and support this intent.

== mosquito, farmers and bio-furturism
. primary medicinal ingredient is artemisinin -> found in Artemisia Annua
. unstable production lead to huge price fluctuations for ACTs

=== natural biosynthesis of artemisinin 
CO2 -> acetyl-CoA -> FPP -> Amorphadiene -> Artemisinic acid -> Dihydroartemisnic acid -> Artemisinin 

image::/images/synbio/Scheme-2-Total-synthesis-of-artemisinin-by-Xu-et-al.png[]

=== engineered synthesis of artemisinin 
image::/images/synbio/engineer-artemisinin.png[]

selection of chassis: important (e.coli which has more scientific research behind it) or an industrially relevant strain or yeast

computation design and simulation

. flux based analysis
. linear programming
. stochastic modelling
. dynamic control, strain optimization, kinetic modelling
. directed evolution rational design (QM/MM)
. ML

=== HP 
. problem analysis
. stakeholder engagement and outreach
. science and math communication
. risk assessment and induction of risk
. policy development

=== wet lab
. DNA
. protein

=== dry lab
. fundamentals of modelling biology
. bioreactors
. computational protein design


=== dec 25

=== molecular cloning 1/3

**genes and parts**

*architecture*

. genes are made of parts: transcriptional and translational elements
. promoter: initiate mRNA production of ORF
. terminator: RNA polymerase must stop
. ORF: RNA/protein of interest
. RBS: initiate protein production
. START/STOP codons designate where the ORF begin and end

image::/images/synbio/mrna.png[]

*genetic symbols*

image::/images/synbio/symbols.png[]

*combinations*

. composite parts: no gene expression
. devices: gene expression
. systems: combination of devices for complex tasks

*logic*

can express boolean logic and truth tables

*control*

. promoter and terminator: affect rate of transcription -> mRNA
. RBS: affect rate of translation -> protein
. degrons: affect rate of translation

*promoters*

. initiate mRNA production of ORF
. strong promoter ->  more protein
. physical basis behind TATA box can affect polymerase to protein mRNA
. mutate TATA box -> affect binding affinity
. binding affinity between RNA polymerase 2 and promoter sequence
. -35 region, -10 region, initiation society

. constitutive: not repressed, always active
. inducible: controlled by operator

image::/images/synbio/promoter.png[]

*TT*

. determine where transcription must stop
.  Rho-independent: formation of hairpin that sterically interrupts RNAP
. Rho-dependent: binding of Rho factor to specific sequences in nascent RNA

*RBS*

. init protein production
. consensus RBS sequence: 5AGGAGG3 -> strength of RBS
. strength of RBS is related to binding affinity between mRNA and ribosome (RBS and RNA component)

image::/images/synbio/rbsbinding.png[]

. RBS calculator: predict RBS and generate new ones

*degrons*

. post-translational effects
. protein degradation tags (control rate of translation by changing protein stability)
. can be engineered or endogenous

*prediction*

. when given pooly annotated gene, must annotate to better understand what you're working with
. look for AGGAGG or TATA
. use online web tools
. bioinformatics


*copy and paste*

. use PCR, polymerase chain reaction and thermocycler

*PCR*

1. small sample of DNA (you want to replicate)
2. primers and nucleotides (allow you to keep building new copies of your target DNA)
3. thermocycling process (increase 30 times)
	1. denature (94-100C). double stranded DNA -> single stranded DNA. allow for the primer to complementary to parts of the DNA you want to ampliy
	2.  annealing phase: lower the temperature only primer can bind and not DNA (forward primer)
	3. elongation phase: polymerase in reaction makes DNA 
4. final extension: much more prolonged (step 3) allow for any unfinished replication finishes

image::/images/synbio/pcr.png[]

. primers ampliy segment of DNA
. can also be designed to give new segment with an overhang (append something to the target DNA)

* pasting*

. assembling and disassembling DNA: hydrogen bonding
. linear dna: fragment
. circular dna: plasmid
	. cloning vector: put dna construct into plasmid for long periods of time (more stable)
	. expression vector: want device or system or be expressed (one promoter inside of them, produce protein for you)
	. 3 major functions: only have bacteria to only have that plasmid that has your dna construct
		1. selection of correct plasmid
		2. replication of plasmid
		3. propagation of gene of interest

. amp: selection marker
. (ori: replication element (origin of replication)) high copy numbers: for cloning/storing dna
. polylinker: insertion of GOI allows cell to replicate plasmid and make more GOI copies (blue colonies if no GOI, white colonies if GOI)

*restriction enzymes*

. MCS is based on restriction-enzyme, ligation-dependent cloning
. restriction enzymes are DNA scissors
	. recognize palindromic sequences

image::/images/synbio/resenzyme.png[]

. MoClo and Golden Gate Assembly
image::/images/synbio/gg.png[]

. RE-LDC and MoClo require ligase to stitch together DNA's phosphate backbone for one contiguous segment 
. ligation independent cloning: gibson is LIC varient (use DNA polymerase) 

*editing*

. stitching PCR: remove small and large chucks of DNA with primers and gel extraction
. self-directed mutagenesis (SDM) for plasmids
. CRISPR!

*cell-free expression: transformation*

. put vector into e.coli 
. to get plasmids into e.coli
. chemical transformation (increase permeability of membrane)


*cell-free expression: concept*

. remove machinary and put in lysate

== dec 26

=== deterministic modelling of genetic circuits: part 1

. ingalls ch 2, 3, 7
. Toggle switch (Garder, et al, Nature, 2000)
. Stochasticity in bacterial gene expression: Elowitz, et al. Science 2002

. part 1: modelling unregulated gene expression
. part 2: modelling regulated gene expression
. part 3: toggle switch
. part 4: stochasticity 
