+++
title = "Commentary: SARS-CoV-2 Evolution: On the Sudden Appearance of the Omicron Variant"
date = 2022-04-28
draft = false
+++

[PDF](https://journals.asm.org/doi/epub/10.1128/jvi.00090-22)

- unusual virus evolution mechanisms: hypermutation, out-of-frame reading, recombination
- regular Darwinian evolution, seems to have led to the altered spike protein of Omicron variant
- Omicron has 60 mutations compared to original Wuhan isolate
  - 35 mutations in gene encoding spike protein => 30 amino acid substitutions
    - unusual cluster of mutational changes: initial mutation caused loss in fitness, which triggered additional but compensatory changes to mitigate fitness loss
    - substitutions may impact cell tropism and pathogenicity
- Omicron has three sublinages:
  - had time to diversify before it was discovered

Darwinian evolution
- lots of selection pressure on the viral spike protein

acquisition of many clustered mutations in spike protein could also have explained from:

hypermutation
- multiple changes occur in clustered manner in parts of the viral genome
- introduced either by viral polymerase or cellular restriction enzymes
- characteristic: common mutational pattern can usually be recognized
- absence of any particular mutational profile does not support the hypermutation scenario

double-frameshift
- ribosome temporally read in different reading frame
- out-of-frame reading would have to be restricted to short segment of the spike protein ORF as extended out-of-frame reading will likely be lethal as inactive proteins could be made or stop codons could be encountered
- but all four indel mutations of Omicron are in frame

recombination
- mixing of Omicron genome sequence with other sequences derived from other viruses or human host
- no good evidence as of now

Possible factors causing Darwinian evolution
- chronic infection in immunocompromised patients is perfect place for SARS-CoV-2 mutations to occur
- selective pressures with vaccination
- viral fitness
- virus transmissability 
