<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8" />
  <meta http-equiv="x-ua-compatible" content="ie=edge" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <title>Lucy H | Paper: Efficiently Enabling Block Semantics and Data Updates in DNA Storage</title>
  <link rel="stylesheet" href="../../../css/main.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/styles/default.min.css" />
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/highlight.min.js"></script>
  <!-- and it's easy to individually load additional languages -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/languages/scheme.min.js"></script>
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/styles/base16/onedark.min.css" />
  <script>
    hljs.highlightAll();
  </script>
<script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
</script>

  <link rel="preconnect" href="https://www.googletagmanager.com" />
  <link rel="preconnect" href="https://www.google-analytics.com" />

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-KDGPVMHC9Q"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag() {
      dataLayer.push(arguments);
    }
    gtag("js", new Date());

    gtag("config", "G-KDGPVMHC9Q");
  </script>

  <!-- basic favicon -->
  <link rel="icon" href="../../../images/android-chrome-384x384.png" />
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="../../../images/android-chrome-384x384.png" />
  <link rel="apple-touch-icon-precomposed" sizes="72x72" href="../../../images/android-chrome-384x384.png" />
  <link rel="apple-touch-icon-precomposed" href="../../../images/android-chrome-384x384.png" />
</head>

  <body>
    <header>
      <div class="logo">
        <a href="../../../">
          <img src="../../../images/android-chrome-384x384.png" style="padding-right: 10px" />
        </a>
      </div>
    </header>

    <main role="main">
      <h1>Paper: Efficiently Enabling Block Semantics and Data Updates in DNA Storage</h1>
      <div>
  
  <div class="header">
    Posted on December 28, 2023
    
  </div>
  <div class="tags">
    
    Tags: <a title="All pages tagged 'synbio'." href="../../../tags/synbio/index.html">synbio</a>, <a title="All pages tagged 'paper'." href="../../../tags/paper/index.html">paper</a>
    
  </div>
  <div class="content">
    <div class="paragraph">
<p><a href="https://arxiv.org/abs/2212.13447">Source</a></p>
</div>
<div class="sect1">
<h2 id="_abstract">abstract</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>storage space into fixed-sized units</p>
</li>
<li>
<p>pair of random access PCR primers of length 20 define an independent storage partition, which is managed indepedently of other partitions</p>
</li>
<li>
<p>transform internal addressing scheme of a partition into an equivalent PCR-compatible → run PCR with primers that can be variably elongated to include desired part of internal address</p>
<div class="ulist">
<ul>
<li>
<p>retrieve a specific block with high accuracy</p>
</li>
<li>
<p>140x reduction in sequencing cost and latency</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_introduction">introduction</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>similarity search</p>
</li>
<li>
<p>runtimes for automating wetlab protocols</p>
</li>
<li>
<p>provide random access at nearly constant latency via PCR</p>
</li>
<li>
<p>current state of the art: key-valye object stores</p>
<div class="ulist">
<ul>
<li>
<p>pair of primers define kay and arbitrarly sized value stored in molecules tagged with same primers</p>
</li>
</ul>
</div>
</li>
<li>
<p>large objects stored across many molecules, all of which are retrieved with sufficient uniformity</p>
<div class="ulist">
<ul>
<li>
<p>internal address: part of molecule that uniquely identify and re-establish order</p>
</li>
<li>
<p>object than spans N molecules requires log_4 N bases for indexing</p>
</li>
</ul>
</div>
</li>
<li>
<p>primers:</p>
<div class="ulist">
<ul>
<li>
<p>lots of unique combinations but all aren’t appropriate</p>
</li>
<li>
<p>must be balanced: GC content (50%)</p>
</li>
<li>
<p>all primers used in same DNA sample must be significantly different from each other in Hamming distance to avoid amplication of unwanted data</p>
<div class="ulist">
<ul>
<li>
<p>minimum pairwise distance (a problem) only around 1000-3000 primers can fit the constraints</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</li>
<li>
<p>can only chemically distinguish 1000 different objects</p>
</li>
<li>
<p>with 1TB of data, the smallest unit of access is 1GB, so we always have to sequence this much DNA, no matter how small the unit we want is</p>
</li>
<li>
<p>using longer primers doesn’t work: compatible primers scale linearly</p>
</li>
<li>
<p>can’t build a functional data storage because of:</p>
<div class="ulist">
<ul>
<li>
<p>allowing arbitrary size, can’t easily get small objects</p>
<div class="ulist">
<ul>
<li>
<p>high Hamming distance</p>
</li>
<li>
<p>many cycles of PCR</p>
</li>
<li>
<p>object-based design results in flat architechture</p>
<div class="ulist">
<ul>
<li>
<p>no logical order/distance metric between objects</p>
</li>
</ul>
</div>
</li>
<li>
<p>limiting sequential access to a single object → random access</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</li>
</ul>
</div>
<div class="paragraph">
<p>paper’s proposal: block storage semantics
- each block independently read/written to
- group of consecutive blocks can be efficieetnly retriveed
- numer of mutually compatible primer pairs is limited, internal address space avalible to any pair of primers is virutally unlimited</p>
</div>
<div class="paragraph">
<p>contributions
- flexibility and constraints of internal address space for pair of primers → implement block storage sementics
- PCR-compatible indexes to enable random access with primers than can be enlongated
- organize partitions similar to version control
- wetlab evaluation with state-of-the-art DNA storage architecture</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_background">background</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_2_1_dna_storage_basics">2.1 dna storage basics</h3>
<div class="ulist">
<ul>
<li>
<p>dna molecule 300 bases long can store 75 bytes</p>
</li>
<li>
<p>have to break down files into shorter molecules, and add an internal address or index</p>
</li>
</ul>
</div>
<div class="sect3">
<h4 id="_2_1_1_encoding">2.1.1 encoding</h4>
<div class="ulist">
<ul>
<li>
<p>using a coding scheme (constrained coding) that</p>
<div class="ulist">
<ul>
<li>
<p>prevent occurrence of homopolyers</p>
</li>
<li>
<p>balance GC content</p>
</li>
</ul>
</div>
</li>
<li>
<p>unconstrained coding: homopolymers occur with low probability</p>
<div class="ulist">
<ul>
<li>
<p>significantly increases coding density with practical ranges of error rates</p>
</li>
<li>
<p>used in this paper</p>
</li>
<li>
<p>handle error types with outer Reed-Solomon ECC codes</p>
</li>
</ul>
</div>
</li>
<li>
<p>to encode internal address within every molecule</p>
<div class="ulist">
<ul>
<li>
<p>use more complex constrained coding scheme</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<div class="imageblock">
<div class="content">
<img src="../../../images/block/fig1.png" alt="fig1">
</div>
</div>
<div class="ulist">
<ul>
<li>
<p>once file split into pieces and encoded into DNA, pair of primers added to end and beginning of each string</p>
<div class="ulist">
<ul>
<li>
<p>chemical tag that logically group related molecules together, allowing for random access</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
<div class="sect3">
<h4 id="_2_1_2_data_retrieval">2.1.2 data retrieval</h4>
<div class="ulist">
<ul>
<li>
<p>PCR used to selectively amplify molecules containing file peices</p>
<div class="ulist">
<ul>
<li>
<p>molecules isolated using primers</p>
</li>
<li>
<p>sequencing produces many DNA strings: reads</p>
</li>
<li>
<p>errors are manifested in reads</p>
</li>
<li>
<p>sequencing coverage/depth: the higher the coverage, the easier to reconstruct from these reads, but higher cost</p>
</li>
</ul>
</div>
</li>
<li>
<p>obtained reads clustered based on similarity (should be from same molecule) via Levenshtein distance</p>
</li>
<li>
<p>consensus finding algorithm to extract original DNA string</p>
</li>
</ul>
</div>
</div>
<div class="sect3">
<h4 id="_2_1_3_decoding_and_error_correction">2.1.3 decoding and error correction</h4>
<div class="ulist">
<ul>
<li>
<p>using internal address, binary data used to recreate original file</p>
</li>
<li>
<p>errors corrected using Reed-Solomon or LDPC codes</p>
</li>
<li>
<p>state of the art architecture treats all DNA molecules as columns in matrix</p>
<div class="ulist">
<ul>
<li>
<p>seperate DNA molecules created as external ECC code, one codeword rerepsent a row → high information density it creates strong inter-molecular dependencies</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
<div class="sect3">
<h4 id="_2_1_4_pcr">2.1.4 PCR</h4>
<div class="ulist">
<ul>
<li>
<p>doubles number of DNA molecules in every cycle</p>
</li>
<li>
<p>three phases: denaturation, annealing, extension</p>
</li>
<li>
<p>optimal primer length: 20</p>
</li>
<li>
<p>can go to 40 and max length 100</p>
</li>
<li>
<p>internal addresses can be very similar, not balanced GC content and repeated base</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_existing_data_update_mechanisms">existing data update mechanisms</h3>
<div class="ulist">
<ul>
<li>
<p>direct edits of DNA molecules: limited to updates to single molecule, can only be applied when size of data does not change</p>
</li>
<li>
<p>current state of the art orgzanizations have data split across many molecules, significant intermolecular data dependencies</p>
</li>
<li>
<p>instead of storing data as nucleotides, store as nicks: rewritable</p>
<div class="ulist">
<ul>
<li>
<p>but cannot do PCR, sacrificing random access</p>
</li>
<li>
<p>storage density is 50-fold lower</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_3_managing_internal_address_space">3 managing internal address space</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>pair of primers of length 20</p>
</li>
<li>
<p>length of DNA strands: 150</p>
</li>
<li>
<p>maximum storage capacity achieved when entire portion of NA used for indexing (no actual data stored)</p>
</li>
<li>
<p>density highest when onle one molecule, no index needed (only for tiny objects)</p>
</li>
<li>
<p>longer primers reduct information density significantly, loss diminishes linearly with longer strand length</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_3_1_parition_architecture">3.1 parition architecture</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>address space AAA…​AAA to TTT…​TTT with index length L represented as prefix tree</p>
<div class="ulist">
<ul>
<li>
<p>index of length L covers addresses from AAA…​AAA to TTT…​TTT: 1D array of 4^L fixed capacity storage units (payload of one molecule)</p>
</li>
<li>
<p>any contigous byte-range can be statically mapped to contigous index-range and vice versa</p>
</li>
</ul>
</div>
</li>
<li>
<p>contiguous index-range be can precisely described with few prefixes: AAA to AGT → AA, AC, AG and AT</p>
</li>
<li>
<p>leaf is DNA strand</p>
</li>
<li>
<p>maximum information density achieved with all DNA strands have same index length and all indexes present</p>
</li>
<li>
<p>any contiguous range of bytes within parition could be retrived quite precisely with single PCR if primers were extended to include part of the index</p>
</li>
<li>
<p>future work: to improve efficiency of sequential accesses: set of files could be mapped onto the partition in a manner that tries to optimally align the files to nodes in prefix tree</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_3_2_concentration_constraints">3.2 concentration constraints</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>to manage cost of sequencing: every strand in DNA storage should be represented in equal concentrations</p>
<div class="ulist">
<ul>
<li>
<p>highly concentrated strands will be sequenced at higher coverage while other strands need more sequencing to be represented, wasting resources</p>
</li>
<li>
<p>for random black access to work: desired sequences need to be amplified much higher because indexes are similar</p>
</li>
<li>
<p>PCR may overwrite their index to desired index, resulting in exponential amplification → mispriming</p>
<div class="ulist">
<ul>
<li>
<p>multiple dna strands, can’t use consensus to select which one</p>
</li>
</ul>
</div>
</li>
<li>
<p>to ensure dominance of desired strands: ensure closet targets in pool are not present in higher concentration</p>
</li>
</ul>
</div>
</li>
<li>
<p>all nodes at same level of index tree map to similar number of DNA strands and in similar concentrations</p>
<div class="ulist">
<ul>
<li>
<p>make sure data updates do not compromise this balance</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_4_random_block_access">4 random block access</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_4_1_general_approach">4.1 general approach</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>indexes AAA…​AAA to TTT…​TTT are not PCR-compatible</p>
</li>
</ul>
</div>
<div class="imageblock">
<div class="content">
<img src="../../../images/block/primer.png" alt="primer">
</div>
</div>
<div class="ulist">
<ul>
<li>
<p>prior work: follow maximum information density design → no control over their structure</p>
</li>
<li>
<p>use less dense encoding of indexes, minor loss in information density</p>
</li>
<li>
<p>added sparsity provide protection against errors in index</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_4_2_requirements_for_elongated_primers">4.2 requirements for elongated primers</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>every pair of main primer: high mutual distance so can extract target partition regardless of its size and concentration relative to other paritions</p>
<div class="ulist">
<ul>
<li>
<p>two similar primers: P_a and P_b</p>
<div class="ulist">
<ul>
<li>
<p>target data is dominant because of uniform concentration of data within same level of index hierarchy</p>
</li>
<li>
<p>ensuring high distance between indexes, is not as important as it is for main primers</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</li>
<li>
<p>GC content: primers have slightly more restrictions</p>
<div class="ulist">
<ul>
<li>
<p>primers are not fixed length: add sparsity to indexes so GC content is uniform with every elongation</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_4_3_pcr_navigable_index_tree">4.3 PCR-Navigable Index Tree</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>prefix tree is randomized (the order of edges): to prevent incomplete, unbalanced, degenerate trees</p>
</li>
<li>
<p>sparse out addresses: add extra letter between every two adjacent edges</p>
<div class="ulist">
<ul>
<li>
<p>perfectly balance the GC content (maximize Hamming distance between sibling nodes)</p>
<div class="ulist">
<ul>
<li>
<p>guatantees near perfect GC content and disable indexes of more than 2</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</li>
<li>
<p>10-based long internal address: 3% information density loss</p>
<div class="ulist">
<ul>
<li>
<p>added flexibility reduces sequencing costs</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_4_4_index_tree_management">4.4 Index Tree Management</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>different seeds for different partitions to ensure vastly different trees to avoid unwanted molecular interactions</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_5_data_updates_in_dna_storage">5 Data Updates in DNA Storage</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>low-latency enzymatic synthesis</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_5_1_naive">5.1 naive</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>throw away old data, update software to just use new primers</p>
</li>
<li>
<p>problems:</p>
<div class="ulist">
<ul>
<li>
<p>recreate parition (expensive, arbitrary amount of information)</p>
</li>
<li>
<p>wasting primres</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_5_2_versioning">5.2 versioning</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>updates logged as ordered series of incremental patches</p>
</li>
<li>
<p>minimal set of DNA molecules at same concentration</p>
</li>
<li>
<p>application of updates done in software, not in-situ chemically</p>
</li>
<li>
<p>how should updates be tagged and their primers?</p>
</li>
<li>
<p>how to retrieve updated DNA and how to mix original DNA with updated DNA</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_5_3_placement_of_updates_in_address_space">5.3 placement of updates in address space</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>updates in their own partition with dedicated set of primers</p>
</li>
<li>
<p>have to potentially read all updates, none of which apply to data (becaues data might have been updated)</p>
</li>
<li>
<p>updates embedded into address space of each primer pair → reading updated data requires single PCR</p>
<div class="ulist">
<ul>
<li>
<p>but have to read all data under same primer pair → lots of data</p>
</li>
</ul>
</div>
</li>
<li>
<p>final: data and updates interwoven</p>
<div class="ulist">
<ul>
<li>
<p>allocate blocks for updates, some unused, overflow data is given a pointer</p>
<div class="ulist">
<ul>
<li>
<p>object with prefix ACGT: original object at ACGTA, first update: ACGTC, second update ACGTG</p>
<div class="ulist">
<ul>
<li>
<p>link between data and updates, no bookkeeping</p>
</li>
</ul>
</div>
</li>
<li>
<p>PCR uses primers with ACGT, software understands how to patch everything together</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_5_4_structure_and_semantics_of_updates">5.4 structure and semantics of updates</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>ensure concentrations of original DNA and updates are similar as possible</p>
</li>
<li>
<p>application of updates delegated to end-user or upper-application</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_5_5_physically_mixing_data_and_updates">5.5 physically mixing data and updates</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>directly impacts cost of sequencing</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_6_methodology_skipped_for_now">6 methodology (skipped for now)</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_7_results">7 results</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_7_1_baseline_random_access">7.1 baseline random access</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>sequencing of 99.66% unwanted data</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_7_2_random_block_access">7.2 random block access</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>mispriming is an issue with similar primers</p>
</li>
<li>
<p>most errors caused by blocks amplified through mispriming</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_7_3_sequencing_cost_reduction">7.3 sequencing cost reduction</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>ability to retrieve individual data blocks</p>
</li>
<li>
<p>sequencing cost is proportional to size of sequencing cost</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_7_4_sequencing_latency_reduction">7.4 sequencing latency reduction</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>ability to select given block, but latency depends on partition size and sequencing technology</p>
<div class="ulist">
<ul>
<li>
<p>block based: retrivign single block reduces number of runs needed</p>
</li>
</ul>
</div>
</li>
<li>
<p>NGS: sequencing output only avalible at end of run</p>
</li>
<li>
<p>nanopore: output size dependent, continuosly produced and analyzed in real time</p>
</li>
<li>
<p>post-sequencing/data movement and software decoding time also reduced (not a bottleneck so doesn’t really matter)</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_7_5_cost_of_creating_and_retrieving_updates">7.5 cost of creating and retrieving updates</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>naive system: new updated copy of entire partition and assigns new primer</p>
</li>
<li>
<p>our system: precise access retrieves data and updates for specific block</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_7_6_mixing_data_and_updates">7.6 mixing data and updates</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>possible to keep concentrations of data and updates similar</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_7_7_scalabilty_and_limitations">7.7 scalabilty and limitations !!</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_7_7_1_block_count">7.7.1 block count</h3>
<div class="ulist">
<ul>
<li>
<p>extend both primers instead of one</p>
<div class="ulist">
<ul>
<li>
<p>two-sided extension by 10 characters → 1024^2 addressable blocks, same order of magnitude as modern SSDs</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_7_7_2_block_size">7.7.2 block size</h3>
<div class="ulist">
<ul>
<li>
<p>mispriming depends on number of blocks and structure and sparsity of their indexes</p>
</li>
<li>
<p>no limits on block size</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_7_73_partition_count">7.73 partition count</h3>
<div class="ulist">
<ul>
<li>
<p>prevent mispriming: two PCR cycles</p>
<div class="ulist">
<ul>
<li>
<p>first main primers</p>
</li>
<li>
<p>second elongated primers</p>
</li>
</ul>
</div>
</li>
<li>
<p>zipfan distrbution: many blocks never accessed, few accessed very frequently</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_8_decoding_procedure">8 decoding procedure</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>extract substring between elongated forward primer and reverse primer (payload)</p>
</li>
<li>
<p>cluster payloads as per Rashtchian</p>
</li>
<li>
<p>in descending order of cluster size: trace reconstruction using double sided BMA algorithm</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_8_1_handing_of_mispriming_during_pcr">8.1 handing of mispriming during pcr</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>incorrectly amplified strands: indexes that were very close to indexes of target block in edit distance (2-3), rather than hamming</p>
</li>
<li>
<p>to an extent, can be corrected through error correction codes</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_9_related_work">9 related work</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>nested PCRs</p>
<div class="ulist">
<ul>
<li>
<p>don’t support multiplex-PCR (this does)</p>
</li>
</ul>
</div>
</li>
<li>
<p>nested PCR is only two levels deep, this is 6 levels deep</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_conclusion">conclusion</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>slow storage media (DNA) so should do update applications in software</p>
</li>
</ul>
</div>
</div>
</div>

  </div>
  <div class="flex-row link-no-style">
    
    <a href="../../../posts/notes/2023-12-29-quantum-mechanics-links/index.html">⇐ Newer post</a>
    
    
    <a href="../../../posts/2023-12-26-not-written-inawhile/index.html">Older post ⇒</a>
    
  </div>
</div>


    </main>

    <footer>
      <center>
        <p>
          <a href="../../../archive">archive</a>
          <a href="https://github.com/lhao03" target="_blank">github</a>
          <a href="https://www.linkedin.com/in/lucy-hao/" target="_blank">linkedin</a>
        </p>
      </center>
      Site powered by cats, 🍫 and
      <a href="http://jaspervdj.be/hakyll">Hakyll</a>
    </footer>
  </body>
</html>
