<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8" />
  <meta http-equiv="x-ua-compatible" content="ie=edge" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <title>Lucy H | Scalable DNA Storage with SynBio</title>
  <link rel="stylesheet" href="../../../css/main.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/styles/default.min.css" />
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/highlight.min.js"></script>
  <!-- and it's easy to individually load additional languages -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/languages/scheme.min.js"></script>
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/styles/base16/onedark.min.css" />
  <script>
    hljs.highlightAll();
  </script>
<script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
</script>

  <link rel="preconnect" href="https://www.googletagmanager.com" />
  <link rel="preconnect" href="https://www.google-analytics.com" />

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-KDGPVMHC9Q"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag() {
      dataLayer.push(arguments);
    }
    gtag("js", new Date());

    gtag("config", "G-KDGPVMHC9Q");
  </script>

  <!-- basic favicon -->
  <link rel="icon" href="../../../images/android-chrome-384x384.png" />
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="../../../images/android-chrome-384x384.png" />
  <link rel="apple-touch-icon-precomposed" sizes="72x72" href="../../../images/android-chrome-384x384.png" />
  <link rel="apple-touch-icon-precomposed" href="../../../images/android-chrome-384x384.png" />
</head>

  <body>
    <header>
      <div class="logo">
        <a href="../../../">
          <img src="../../../images/android-chrome-384x384.png" style="padding-right: 10px" />
        </a>
      </div>
    </header>

    <main role="main">
      <h1>Scalable DNA Storage with SynBio</h1>
      <div>
  
  <div class="header">
    Posted on December 31, 2023
    
  </div>
  <div class="tags">
    
    Tags: <a title="All pages tagged 'synbio'." href="../../../tags/synbio/index.html">synbio</a>, <a title="All pages tagged 'paper'." href="../../../tags/paper/index.html">paper</a>
    
  </div>
  <div class="content">
    <div class="sect1">
<h2 id="_slide_1">Slide 1</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_slide_2">Slide 2</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_slide_3">Slide 3</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Cloud storage is ubiquitous. In addition to cloud storage data centers operated by technology companies and data-critical (healthcare, finance, and government) organizations, these entities must deal with a growing reliance on storing a lot of information that is rarely changed but required to be stored for long periods of time. This is referred to as an information explosion. We are predicted to generate around 180 zettabytes of information, with 22 zettabytes of data being shipped around, 60% in the form of hard drives.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_slide_4">Slide 4</h2>
<div class="sectionbody">
<div class="paragraph">
<p>What is long term storage? Long term storage is defined as storage that can persist information without the requirement for constant electricity. Silicon based long term storage includes: external hard drives, flash drives, CDs which last around 5-20 years. Silicon based storage works broadly like this: file is turned into a stream of bits of 0s or 1s, and these bits are stored as charges based on the binary encoding, then after these bits have been encoded, the storage media is stored in conditions that try to preserve this charge. Ubiquitous storage in the form of "the cloud" is becoming an everyday norm because most people do not wanna lug around pieces of storage. But cloud storage isn‚Äôt really in the cloud. Cloud storage is stored in giant buildings full of hard drives, some buildings taking up "nearly 200 acres of land apiece". In addition, because Google and Microsoft promise that your data is preserved, multiple copies of your data is written to different data centers. Multiply that process a few times, and you can see that cloud data centers get hot, which requires lots of cooling, resulting in data centers using as much energy as an airline. It‚Äôs not that the actual storage that requires so much cooling, but the computers that write all that data to disk which must be cooled. While these data centers account for around 2% of global energy usage, and because information is growing rapidly, these data centers will continue to eat up more land and energy. Generally speaking, data centers have reached their peak efficiency with the level of scalability require, which we can implicitly observe from Moore‚Äôs Law.</p>
</div>
<div class="paragraph">
<p>DNA storage is an emerging technology for long term storage. Generally the steps of storing information into DNA goes as follows: are encoding, synthesis, storage, retrieval, and decoding. Compared to traditional long term storage mediums, DNA is more:</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>durable: if stored in optimal conditions and dehydrated, DNA can possibly endure for millions of years.</p>
</li>
<li>
<p>dense: up to 10^{19} bits per cubic centimeter, which is eight times more dense than other storage media.</p>
</li>
<li>
<p>energy efficient: due to the failure rate of current long term storage media, such as charges flipping, I predict less energy will be needed to store information in DNA molecules because we can store less copies of data given DNA‚Äôs density. In fact, there are estimates that "it could reduce the energy consumption by 1,000 times compared to today‚Äôs data centres"<sup class="footnote">[<a id="_footnoteref_1" class="footnote" href="#_footnotedef_1" title="View footnote.">1</a>]</sup>.</p>
</li>
</ol>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_slide_3_2">Slide 3</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Currently DNA storage the most durable and dense storage medium we have. However, it is not a easily scalable technology, which is what I believe synthetic biology can help with. Current industry standards of DNA synthesis is done through chemical means, phosphoramidite synthesis. Phosphoramidite synthesis is expensive and requires the use of toxic organic solvents. With sequencing, because DNA is so dense, if implemented with a naive storage model, it‚Äôs easy for too much data to be sequenced because we may have to sequence an entire DNa molecule to get at a few bases in the middle of the molecule. This results in the output from sequencing to be wasted. And lastly, because mutating DNA in the wet lab is prone to mistakes and slower than traditional long term storage, this is another roadblock in realizing DNA as a long term storage medium.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_slide_4_solution">Slide 4: Solution</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Thus, because synthesis and sequencing remain as bottlenecks for using DNA as storage, our solution is to design a scalable DNA storage process with improved synthesis and storage.</p>
</div>
<div class="paragraph">
<p>An emerging method of DNA synthesis is enzymatic DNA synthesis with the use of TdT (template-independent polymerase). Aachen 2020 successfully demonstrated the use of TdT to synthesize DNA. We can go further to parallelize this process with an microelectrode array; this will be driven by wet lab.</p>
</div>
<div class="paragraph">
<p>To improve storage, our dry lab will implement block based semantics with elongated primers and an update system with "patches". Dry lab will also either write software or use open source code to encode and decode information into bases and deal with error correction, based on time constraints.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_slide_5_why_enzymatic_dna_synthesis">Slide 5: Why enzymatic DNA synthesis?</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>terminal deoxynucleotidyl transferase (TdT) is the enzyme behind enzymatic DNA synthesis:</p>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>does not require preexisting strand of DNA</p>
</li>
<li>
<p>Uses soft chemicals</p>
</li>
<li>
<p>Provides kinetic control with enzymes</p>
</li>
<li>
<p>produce longer strands than with chemical synthesis<sup class="footnote">[<a id="_footnoteref_2" class="footnote" href="#_footnotedef_2" title="View footnote.">2</a>]</sup></p>
</li>
</ol>
</div>
</li>
<li>
<p>of course some downsides include that different bases add differently, which is an issue Aachen tackled.</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_slide_6_why_block_based_semantics_and_updates_via_patches">Slide 6: Why block based semantics and updates via patches?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A flat key-value object based architecture forces the user to sequence an entire unit of random access, whether they want all that information or not. This results in a lot of waste because DNA is very dense.</p>
</div>
<div class="paragraph">
<p>With block based semantics:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>we design primers designed with main prefix and extended bases representing blocks:</p>
<div class="ulist">
<ul>
<li>
<p>retrieve a block in a partition with higher accuracy</p>
</li>
<li>
<p>blocks can be smaller than the unit of random access: sequencing smaller blocks of data</p>
</li>
<li>
<p>reductions in sequencing cost and latency</p>
</li>
<li>
<p>ability for sequential access by elongating a primer, similarly done in array based storage (tradditional storage mediums)</p>
</li>
<li>
<p>primers based on tree based structure</p>
</li>
</ul>
</div>
</li>
<li>
<p>versioning: only write updates as new DNA molecule and tag with appropriate primer</p>
<div class="ulist">
<ul>
<li>
<p>updates applied in software is: faster, more reliable</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_wet_lab">Wet lab</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>based on Aachen‚Äôs work, we can first set up the enzymatic DNA synthesis process as described by Church‚Äôs paper.</p>
</li>
<li>
<p>then to demonstrate improvement, we work on creating the novel, cleavable polymerase-nucleotide conjugate which allows for attachment to the microelectrode array, while maintaining the activity of the TdT enzyme.</p>
<div class="ulist">
<ul>
<li>
<p>this allows for electrochemical cleavage/deblocking.</p>
</li>
<li>
<p>compared to parallelized methods that rely on kinetic competitions, polymerase-nucleotide conjugate allows for single-base additions and the ability to further extend the oligonucleotide.</p>
</li>
<li>
<p>electrochemical control vs. nitrous acid-mediated deblocking also means we don‚Äôt rely on environmentally hazardous chemicals</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_dry_lab">Dry Lab</h2>
<div class="sectionbody">
<div class="paragraph">
<p>We can employ the engineering cycle with our software algorithms, which we can referred to as the software codec.</p>
</div>
<div class="paragraph">
<p>There needs to be a process for:
1. Encoding
* requires generating primers to implement the block based semantics (via Hamming/Edit distance constraints and chemistry constraints like GC content) and splitting a file across several DNA molecules if necessary</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>Decoding</p>
<div class="ulist">
<ul>
<li>
<p>error correction, reconstructing the file and applying update patches</p>
</li>
</ul>
</div>
</li>
<li>
<p>Updating data</p>
<div class="ulist">
<ul>
<li>
<p>because we aren‚Äôt applying updates chemically, instead reading a DNA molecule gives us the original information and updates, which with long term storage shouldn‚Äôt be alot. Using algorithms we can apply these updates to produce the final version of the data with updates applied.</p>
</li>
</ul>
</div>
</li>
</ol>
</div>
<div class="paragraph">
<p>For Modelling, we can model:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>the TdT enzyme using Michals-Menten kinetics</p>
</li>
<li>
<p>how likely a primer is to "misprime" meaning the wrong primer is amplified, with protein modelling.</p>
</li>
<li>
<p>optimal concentrations of DNA molecules are required to keep sequencing costs low and</p>
<div class="ulist">
<ul>
<li>
<p>this requires a feedback loop of wet lab giving us data from "Amplify-then-Measure" then dry lab using these numbers to estimate the costs of sequencing based on concentrations</p>
</li>
</ul>
</div>
</li>
<li>
<p>optimal conditions of added different bases</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_human_practices">Human Practices</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This project is very much interdisciplinary, so we must reach to many experts. Our wet lab portion is also chemistry heavy, so I have listed to chemistry professors who I believe will be very helpful.</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Specialists:</p>
<div class="ulist">
<ul>
<li>
<p>Electrochemistry: Dr. Eva Nichols (UBC)</p>
</li>
<li>
<p>Synthetic DNA: Dr. David Perrin (UBC)</p>
</li>
<li>
<p>Data management: Dr. Richard Arias-Hern√°ndez (UBC)</p>
</li>
<li>
<p>Bioinformatics (Error correction): Dr. Jiarui Ding (UBC)</p>
</li>
</ul>
</div>
</li>
<li>
<p>Outreach</p>
<div class="ulist">
<ul>
<li>
<p>Education:</p>
<div class="ulist">
<ul>
<li>
<p>Computation and Information</p>
</li>
<li>
<p>DNA Storage</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</li>
<li>
<p>BC Government, Hospitals</p>
</li>
<li>
<p>Cloud storage companies</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_sources_and_notes">Sources and Notes</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_current_long_term_storage">Current Long Term Storage</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://en.wikipedia.org/wiki/Zettabyte_Era" class="bare">https://en.wikipedia.org/wiki/Zettabyte_Era</a></p>
</li>
<li>
<p><a href="https://mit-serc.pubpub.org/pub/the-cloud-is-material/release/1" class="bare">https://mit-serc.pubpub.org/pub/the-cloud-is-material/release/1</a></p>
</li>
<li>
<p><a href="https://contenteratechspace.com/6-best-ways-for-long-term-data-storage/" class="bare">https://contenteratechspace.com/6-best-ways-for-long-term-data-storage/</a></p>
</li>
<li>
<p><a href="https://medium.com/stanford-magazine/carbon-and-the-cloud-d6f481b79dfe" class="bare">https://medium.com/stanford-magazine/carbon-and-the-cloud-d6f481b79dfe</a></p>
</li>
<li>
<p><a href="https://www.cbc.ca/radio/spark/digital-data-has-an-environmental-cost-calling-it-the-cloud-conceals-that-researcher-says-1.6641268" class="bare">https://www.cbc.ca/radio/spark/digital-data-has-an-environmental-cost-calling-it-the-cloud-conceals-that-researcher-says-1.6641268</a></p>
</li>
<li>
<p><a href="https://thereader.mitpress.mit.edu/the-staggering-ecological-impacts-of-computation-and-the-cloud/" class="bare">https://thereader.mitpress.mit.edu/the-staggering-ecological-impacts-of-computation-and-the-cloud/</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_dna_storage">DNA storage</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10296570" class="bare">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10296570</a></p>
</li>
<li>
<p><a href="https://www.computer.org/csdl/magazine/co/2023/07/10154188/1O1wUGjK41O" class="bare">https://www.computer.org/csdl/magazine/co/2023/07/10154188/1O1wUGjK41O</a></p>
</li>
<li>
<p><a href="https://www.nature.com/articles/d42473-022-00319" class="bare">https://www.nature.com/articles/d42473-022-00319</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_wet_lab_2">Wet lab</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.nature.com/articles/s41467-019-10258-1">Terminator-free template-independent enzymatic DNA synthesis for digital information storage</a></p>
</li>
<li>
<p><a href="https://2021.igem.org/Team:Aachen" class="bare">https://2021.igem.org/Team:Aachen</a></p>
</li>
<li>
<p><a href="https://pubs.acs.org/doi/10.1021/acssynbio.3c00044">Spatially Selective Electrochemical Cleavage of a Polymerase-Nucleotide Conjugate</a></p>
</li>
<li>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8288405/" class="bare">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8288405/</a></p>
</li>
<li>
<p>this approach is chemistry heavy, so based on the skills of our wet lab, we can also engineer yeast cells to produce DNA and store data in artifical chromosomes, as demonstrated by Yuan<sup class="footnote">[<a id="_footnoteref_3" class="footnote" href="#_footnotedef_3" title="View footnote.">3</a>]</sup></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_dry_lab_2">Dry Lab</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.nature.com/articles/srep14138" class="bare">https://www.nature.com/articles/srep14138</a></p>
</li>
<li>
<p><a href="https://arxiv.org/abs/2212.13447" class="bare">https://arxiv.org/abs/2212.13447</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div id="footnotes">
<hr>
<div class="footnote" id="_footnotedef_1">
<a href="#_footnoteref_1">1</a>. <a href="https://www.nature.com/articles/d42473-022-00319-3" class="bare">https://www.nature.com/articles/d42473-022-00319-3</a>
</div>
<div class="footnote" id="_footnotedef_2">
<a href="#_footnoteref_2">2</a>. <a href="https://www.forbes.com/sites/johncumbers/2022/03/25/dna-synthesis-goes-green/?sh=49f785fb532c" class="bare">https://www.forbes.com/sites/johncumbers/2022/03/25/dna-synthesis-goes-green/?sh=49f785fb532c</a>
</div>
<div class="footnote" id="_footnotedef_3">
<a href="#_footnoteref_3">3</a>. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8288405/" class="bare">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8288405/</a>
</div>
</div>

  </div>
  <div class="flex-row link-no-style">
    
    <a href="../../../posts/notes/2024-02-03-repmi/index.html">‚áê Newer post</a>
    
    
    <a href="../../../posts/notes/2023-12-30-combinatorial-pcr-method/index.html">Older post ‚áí</a>
    
  </div>
</div>


    </main>

    <footer>
      <center>
        <p>
          <a href="../../../archive">archive</a>
          <a href="https://github.com/lhao03" target="_blank">github</a>
          <a href="https://www.linkedin.com/in/lucy-hao/" target="_blank">linkedin</a>
        </p>
      </center>
      Site powered by cats, üç´ and
      <a href="http://jaspervdj.be/hakyll">Hakyll</a>
    </footer>
  </body>
</html>
